json-format.md

```diff
@@ -157,9 +156,7 @@ Example event with `Binary`-valued data
     "source" : "/mycontext",
     "eventID" : "B234-1234-1234",
     "eventTime" : "2018-04-05T17:31:00Z",
-    "extensions" : {
-      "comExampleExtension" : "value"
-    },
+    "comExampleExtension" : "value",
```

Comment:
Previously created messages can suddenly have undefined behaviors if they collide with this extension. At best, the colliding value is of a different type (like a `Map`), so the previously valid message is now malformed. At worst, the colliding value is of the same type (`String`) so the reader has no idea that the input is garbage, which leads to undefined behavior.

Granted, even if we confine extensions to a separate bag we can still have collisions. But extensions are understood to be a testing ground that comes with risks, and people can choose to ignore any extension and still have basic functionality. With this PR, we are encouraging people to put arbitrary key/values at the top level with the caveat that at any time the CloudEvent spec may define an official interpretation that's different from theirs. I think restricting extensions to a separate bag would minimize this sort of problem.

Reply:
Correct, collisions can happen but no one has expressed a desire to add a ton more properties to the spec - a few perhaps, but not a lot, so its still a pretty small chance.  I believe that most 3rd party extensions will have names that are going to be either scoped by some unique part (e.g. `company-` prefix) or the purpose will be so specific to a usecase that the odds of it being picked up as "generic" and part of the spec are slim - and thus its name should be pretty specific as well.

As for a generic extension, my hope is that people bring those to the WG so we can work on them in our extensions doc.  I view this as a similar potential problem as HTTP headers. Sure extension HTTP headers could have collisions with future HTTP spec defined headers but in practice it hasn't really been an issue.

Reply:
I am not sure if this is the best place to add a classification bag example.  But to avoid people's confusion that only key-value pair format is allowed in the main spec.md and extensions.md, please add the following to this example:
    “identityLabels” : {                      	
          …
    }

Reply:
I'll add a complex type extension to make it clear its not just simple name/value pairs.